src/main/java/uk/gov/hmcts/ccd/data/ReferenceDataRepository.java

```diff
@@ -0,0 +1,105 @@
+package uk.gov.hmcts.ccd.data;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.ccd.ApplicationParams;
+import uk.gov.hmcts.ccd.domain.model.refdata.BuildingLocation;
+import uk.gov.hmcts.ccd.domain.model.refdata.Service;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Named
+@Slf4j
+public class ReferenceDataRepository {
+    private final SecurityUtils securityUtils;
+    private final RestTemplate restTemplate;
+    private final ApplicationParams applicationParams;
+    private final CacheManager cacheManager;
+
+    private static final String BUILDING_LOCATIONS_CACHE = "buildingLocations";
+    private static final String SERVICES_CACHE = "orgServices";
+    private static final String CACHE_KEY = "#root.method.name";
+
+    static final String BUILDING_LOCATIONS_PATH = "/refdata/location/building-locations";
+    static final String SERVICES_PATH = "/refdata/location/orgServices";
+
+    @Inject
+    public ReferenceDataRepository(final SecurityUtils securityUtils,
+                                   @Qualifier("restTemplate") final RestTemplate restTemplate,
+                                   final ApplicationParams applicationParams,
+                                   final CacheManager cacheManager) {
+        this.securityUtils = securityUtils;
+        this.restTemplate = restTemplate;
+        this.applicationParams = applicationParams;
+        this.cacheManager = cacheManager;
+    }
+
+    @Cacheable(cacheNames = BUILDING_LOCATIONS_CACHE, key = CACHE_KEY)
+    public List<BuildingLocation> getBuildingLocations() {
+        return getReferenceData(BUILDING_LOCATIONS_PATH, BuildingLocation[].class);
+    }
+
+    @Cacheable(cacheNames = SERVICES_CACHE, key = CACHE_KEY)
```

Comment:
can we add `unless` condition to the annotation avoid caching of empty or null List? 
`unless = "#result==null or #result.isEmpty()"`